web/modules/custom/dpl_event/src/ReoccurringDateFormatter.php

```diff
@@ -90,19 +90,12 @@ public function getSeriesDateString(EventSeries $event_series): string|null {
   }
 
   /**
-   * Retrieves the upcoming event details for a given event series.
-   *
-   * @param \Drupal\recurring_events\Entity\EventSeries $event_series
-   *   The event series object.
+   * Finding the IDs of all events that are in the future.
    *
-   * @return null|array{'start': \Drupal\Core\Datetime\DrupalDateTime, 'end': \Drupal\Core\Datetime\DrupalDateTime, 'upcoming_ids': array<int, string>}
-   *   An array containing the following keys:
-   *   - start: The start date of the upcoming event as a DrupalDateTime object.
-   *   - end: The end date of the upcoming event as a DrupalDateTime object.
-   *   - upcoming_ids: An array of upcoming event instance IDs.
-   *   Returns NULL if the start and end dates are not found.
+   * @return array<int, string>
+   *   An array of matching eventinstance IDs.
    */
-  public function getUpcomingEventDetails(EventSeries $event_series): array|null {
+  public function getUpcomingEventIds(EventSeries $event_series): array {
```

Comment:
In general the method would be easier to use for callers if it returned an array of `EventInstance` objects.

One way to achieve this would be to look at `$event_series->get('event_instances')->referencedEntities()` and filter away the ones with a date in the past.

Since this is a field value I would expect this to be at least as efficient.

Reply:
I'd like to avoid doing that, because technically, it could end up loading a *lot* of entities, for no reason, that we're gonna filter out anyway.
referencedEntities loads the whole entity

Reply:
OK. Let's leave it as is for now.